### PR TITLE
Add GH Issue forms and Pull Request template

### PR DESCRIPTION
**Pull request by connortann**
Supports #2980.

For reference, see  https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/

Adapted from / inspired by: https://github.com/pandas-dev/pandas/tree/main/.github/ISSUE_TEMPLATE

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (thatlittleboy) [inline .github/PULL_REQUEST_TEMPLATE.md]: The "Replace xxxx ..." comment can be an HTML comment, so it doesn't show up in the PR description if the OP doesn't remove it.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (thatlittleboy) [inline .github/PULL_REQUEST_TEMPLATE.md]: I don't think this is necessary. Us maintainers can enforce it just by making the tests required before the merge.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (thatlittleboy) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'll also enforce that the OP provides a PR description, to give a summary of the changes for easier review.

Copying this from one of my other projects:

```
# PR Description

Please describe the changes proposed in the pull request:

-
-
```

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (thatlittleboy) [inline .github/ISSUE_TEMPLATE/feature-request.yml]: pseudo

File: .github/ISSUE_TEMPLATE/feature-request.yml
Comment (thatlittleboy) [inline .github/ISSUE_TEMPLATE/feature-request.yml]: I think we can also add one more section (checkbox) at the end asking if the requester is willing to assist in implementing the feature. Just an idea.

File: .github/ISSUE_TEMPLATE/bug-report.yml
Comment (thatlittleboy) [inline .github/ISSUE_TEMPLATE/bug-report.yml]: The link can be updated to `https://matthewrocklin.com/minimal-bug-reports` to save on one level of redirection.

**Review by thatlittleboy — CHANGES_REQUESTED** [review]
Sorry for the delay! Just a couple of comments for your consideration

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (connortann) [reply .github/PULL_REQUEST_TEMPLATE.md]: I wasn't thinking about enforcement, but rather efficiency.

Along the principle of [shift-left testing](https://en.m.wikipedia.org/wiki/Shift-left_testing), my thinking is that we can make the PR process more seamless by encouraging contributors to run the linter from the get-go. This saves contributors time (debugging failed Checks, making fixup commits), and maintainers time (on code review comments).

**Comment by connortann** [comment]
Thanks for the thorough review! I'm AFK this weekend but will aim to fixup on Monday.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (thatlittleboy) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yup fair point.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (connortann) [reply .github/PULL_REQUEST_TEMPLATE.md]: Good idea, added.

**Review by thatlittleboy — APPROVED** [review]
joey-friends-nice

**Pull request merged**
